### ISSUE TITLE
CUDA Messages in TensorFlow 2.17.0 and tf-nightly

### ISSUE THREAD

**Issue by arthurflor23** [tensorflow/tensorflow#71791]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.17.0

### Custom code

Yes

### OS platform and distribution

Linux

### Python version

3.12

### Current behavior?

Hi,

I've updated TensorFlow to version 2.17.0 and noticed that several CUDA-related messages appear during model training.

Example messages:
```
I0000 00:00:1720821406.093527 2748382 cuda_executor.cc:821] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
I0000 00:00:1720821406.127659 2748382 cuda_executor.cc:821] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
I0000 00:00:1720821406.127833 2748382 cuda_executor.cc:821] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
I0000 00:00:1720821406.161631 2748382 cuda_executor.cc:821] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
```

```
I0000 00:00:1720821413.362793 2748566 cuda_dnn.cc:530] Loaded cuDNN version 8907
W0000 00:00:1720821413.423013 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.441550 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.442263 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.447093 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.447949 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.449831 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.450692 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.451745 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.452589 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
W0000 00:00:1720821413.453438 2748566 gpu_timer.cc:114] Skipping the delay kernel, measurement accuracy will be reduced
[...]
```

I receive numerous messages of 'Skipping the delay kernel, measurement accuracy will be reduced' at the beginning of training. The messages don't affect the training process, however, are new and appear only in the recent version (2.17.0 and tf-nightly).

The flag `TF_CPP_MIN_LOG_LEVEL` doesn't make any difference, so any assistance in understanding or suppressing these messages would be appreciated.

### Standalone code to reproduce the issue

When importing tensorflow it is already possible to see some messages of this type.

```
python -c "import os; os.environ['TF_CPP_MIN_LOG_LEVEL']='3'; import tensorflow as tf;"

WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1720831683.868288 3059140 cuda_dnn.cc:8458] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1720831683.872386 3059140 cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
```

**Comment by tilakrayal** [comment]
@arthurflor23,
Thank you for reporting the issue. This is a known issue where other issues are still open and developers are working on the same. 

I request you to take a look at this [issue](https://github.com/tensorflow/tensorflow/issues/62075) and where a similar issue has been proposed and it is still open. Also I request to follow the similar issue which has been proposed to have the updates on the similar issue.

https://github.com/tensorflow/tensorflow/issues/70947

Thank you!

**Comment by tourist-C** [comment]
Hi, same issue here with model.fit()

I am also getting this log message

```
2024-07-18 10:30:32.699962: I tensorflow/core/framework/local_rendezvous.cc:404] Local rendezvous is aborting with status: OUT_OF_RANGE: End of sequence
	 [[{{node MultiDeviceIteratorGetNextFromShard}}]]
	 [[RemoteCall]]
```

Is there any implication to training speed and training accuracy?

Appreciate your work thanks!

**Comment by tilakrayal** [comment]
cc @belitskiy

**Comment by mmtrebuchet** [comment]
The "skipping the delay kernel" is printed a ridiculous number of times for me - one of my training runs spat out this message 4,428 times. I encourage my users to check their logs for warnings and this sort of behavior teaches users that it's safe to ignore warnings since they don't mean anything. This then leads people to ignore warnings from my code like 
`WARNING: Non-integer read count found. Data should not be normalized when training!` and `WARNING: Divide by zero in seqlet distribution!`, and these are things that indicate that something is possibly wrong.

This is after I run 

```
import os
import warnings
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
import tensorflow as tf
tf.get_logger().setLevel("ERROR")
warnings.simplefilter("ignore")
```

Those messages still get printed, as does

`WARNING: All log messages before absl::InitializeLog() is called are written to STDERR`

and a few XLA messages.

**Comment by Zyrin** [comment]
The "skipping the delay kernel" appears to be new with TensorFlow 2.17.0. And it even appears in some official documentation.
Here is a screenshot from https://www.tensorflow.org/guide/function:
![grafik](https://github.com/user-attachments/assets/5530c1e7-71ee-4484-998c-1fcfadd930ce)
https://www.tensorflow.org/tutorials/quickstart/advanced is also affected.

A warning spammed a gazillion times with no apparent way to fix it is more ore less useless.

**Comment by johndpope** [comment]
resolved in nightly 2.19 
`python3 -m pip install tf-nightly`
